### PR TITLE
feature: implemented waiting for opponent ui

### PR DESCRIPTION
**Pull request by ssen0027**
- Added the waiting for opponent UI.
- Added component for the loading effect spinner.
- Added component that combines the PlayerNamesMobile, StarsInfo and SpectatorCount components, so you don't have to keep typing it all out.
- Added component for displaying the rock, paper, scissors hands (in semi-transparent, solid filled and outlined options). It also animates into the screen when screen is loaded/ refreshed (note: if you're using the inspect/ developer tools to view different screen sizes and keep spam refreshing the page, there might be a bit of a glitch :( I couldn't find anything wrong with the code, but it should work perfectly fine on its first load).

### CODE REVIEW

File: client/src/components/PlayerMoveHand.tsx [hunk 1]
```diff
@@ -0,0 +1,83 @@
...
+import semiTransparentScissors from "../assets/SemiTransparentScissors.svg";
+import filledPaper from "../assets/Paper.svg";
+import filledRock from "../assets/Rock.svg";
+import filledScissors from "../assets/Scissors.svg";
+import outlinedPaper from "../assets/Paper-Outline.svg";
+import outlinedRock from "../assets/Rock-Outline.svg";
+import outlinedScissors from "../assets/Scissors-Outline.svg";
+
+type PlayerMoveHandProps = {
+  playerMove: string; // eg. rock, paper or scissors
```
Comment (RoryT-U) [inline client/src/components/PlayerMoveHand.tsx:13]: Probably want to change this to the Action type in the types.ts.

File: client/src/components/PlayerMoveHand.tsx [hunk 2]
```diff
@@ -0,0 +1,83 @@
...
+  isOpponent: boolean;
+  handType: string; // eg. filled, outlined or semiTransparent
```
Comment (RoryT-U) [inline client/src/components/PlayerMoveHand.tsx:15]: Might also want to make this a type or enum.

File: client/src/components/PlayerMoveHand.tsx [hunk 3]
```diff
@@ -0,0 +1,83 @@
...
+                        : handType === "semiTransparent" &&
+                            playerMove === "rock"
+                          ? semiTransparentRock
+                          : handType === "semiTransparent" &&
+                              playerMove === "paper"
+                            ? semiTransparentPaper
+                            : handType === "semiTransparent" &&
+                                playerMove === "scissors"
+                              ? semiTransparentScissors
+                              : ""
```
Comment (RoryT-U) [inline client/src/components/PlayerMoveHand.tsx:66]: Might be a better way todo this. But I do not currently have any ideas.

**Review by RoryT-U — CHANGES_REQUESTED** [review]
Looks good, some small problems. Also I think Anaad PR #46 also has some of the functionalties that you have implemented, so you two might need to make sure nothing is conflicting.

File: client/src/components/PlayerMoveHand.tsx (hunk 1 above)
Comment (RoryT-U) [reply client/src/components/PlayerMoveHand.tsx:13]: This refers to playerMove attribute.

File: client/src/components/PlayerMoveHand.tsx (hunk 2 above)
Comment (RoryT-U) [reply client/src/components/PlayerMoveHand.tsx:15]: This refers to handType attribute.

File: client/src/components/PlayerMoveHand.tsx (hunk 3 above)
Comment (ssen0027) [reply client/src/components/PlayerMoveHand.tsx:66]: yea i couldn't really come up with a better way to do it ^^; i'll keep thinking tho

**Review by RoryT-U — APPROVED** [review]
Looks great.

**Pull request merged**